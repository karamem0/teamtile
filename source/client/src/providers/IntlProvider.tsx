//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import {
  IntlProvider as Provider,
  createIntl,
  MessageFormatElement
} from 'react-intl';

import en from '../translations/compiled/en.json';
import ja from '../translations/compiled/ja.json';

const translations: { [key: string]: Record<string, MessageFormatElement[]> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language.substring(0, 2)]
});

function IntlProvider(props: React.PropsWithChildren<unknown>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
