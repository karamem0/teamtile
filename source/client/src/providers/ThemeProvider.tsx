//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { Provider, themeNames } from '@fluentui/react-teams';

import { app } from '@microsoft/teams-js';

import { inTeams } from '../utils/Teams';

interface ThemeProviderProps {
  children?: React.ReactNode
}

function ThemeProvider(props: ThemeProviderProps) {

  const { children } = props;

  const [ lang, setLang ] = React.useState<string>('en-US');
  const [ themeName, setThemeName ] = React.useState<themeNames>(themeNames.Default);

  const handleThemeChange = (value: string) => {
    switch (value) {
      case 'dark':
        setThemeName(themeNames.Dark);
        break;
      case 'contrast':
        setThemeName(themeNames.HighContrast);
        break;
      default:
        setThemeName(themeNames.Default);
        break;
    }
  };

  React.useEffect(() => {
    if (!inTeams()) {
      return;
    }
    (async () => {
      await app.initialize();
      const context = await app.getContext();
      setLang(context.app.locale);
      handleThemeChange(context.app.theme);
      app.registerOnThemeChangeHandler(handleThemeChange);
    })();
  }, []);

  return (
    <Provider
      lang={lang}
      themeName={themeName}>
      {children}
    </Provider>
  );

}

export default ThemeProvider;
