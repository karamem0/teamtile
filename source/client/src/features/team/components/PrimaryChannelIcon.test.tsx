//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './PrimaryChannelIcon.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';
import { render } from '@testing-library/react';

it('should create a shapshot when the primary parameter is true', () => {
  const params = {
    primary: true
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should create a shapshot when the primary parameter is false', () => {
  const params = {
    primary: false
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should create a shapshot when the type parameter is undefined', () => {
  const params = {
    primary: undefined
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
