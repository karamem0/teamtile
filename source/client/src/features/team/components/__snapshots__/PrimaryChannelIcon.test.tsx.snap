// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a shapshot when the primary parameter is false 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

<div
    class="fui-FluentProvider fui-FluentProviderr2 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    />
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the primary parameter is true 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #707070;
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    >
      <span
        aria-label="Primary channel"
        class="fui-Text emotion-1 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.31 1.26a1 1 0 0 1 1.38 0l4.84 4.58c.3.28.47.68.47 1.1v5.56c0 .83-.67 1.5-1.5 1.5h-2A1.5 1.5 0 0 1 9 12.5V10a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v2.5c0 .83-.67 1.5-1.5 1.5h-2A1.5 1.5 0 0 1 2 12.5V6.93c0-.41.17-.8.47-1.09L7.3 1.26ZM8 2 3.16 6.57a.5.5 0 0 0-.16.36v5.57c0 .28.22.5.5.5h2a.5.5 0 0 0 .5-.5V10c0-.83.67-1.5 1.5-1.5h1c.83 0 1.5.67 1.5 1.5v2.5c0 .28.22.5.5.5h2a.5.5 0 0 0 .5-.5V6.93a.5.5 0 0 0-.16-.36L8 1.99Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the type parameter is undefined 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

<div
    class="fui-FluentProvider fui-FluentProviderr3 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    />
  </div>
</DocumentFragment>
`;
